src/components/RegisterForm.vue

```diff
@@ -0,0 +1,276 @@
+					Local number
+				</label>
+			</div>
+		</div>
+		<div class="field mb-4 flex justify-content-between">
+			<div class="p-float-label w-10">
+				<div class="p-float-label w-12 pr-1">
+					<InputText
+						id="city"
+						v-model="v$.address.city.$model"
+						:class="{ 'p-invalid': v$.address.city.$invalid && submitted }" />
+					<label for="city">City*</label>
+				</div>
+				<span v-if="v$.address.city.$error && submitted">
+					<span id="city-error" v-for="(error, index) of v$.address.city.$errors" :key="index">
+						<small class="p-error">{{ error.$message }}</small>
+					</span>
+				</span>
+				<small
+					v-else-if="(v$.address.city.$invalid && submitted) || v$.address.street.$pending.$response"
+					class="p-error">
+					{{ v$.address.city.required.$message.replace("Value", "City") }}
+				</small>
+			</div>
+			<div class="p-float-label w-4">
+				<div class="p-float-label w-12 pl-1">
+					<InputMask
+						mask="99-999"
+						id="zipCode"
+						v-model="v$.address.zipCode.$model"
+						:class="{ 'p-invalid': v$.address.zipCode.$invalid && submitted }" />
+					<label for="zipCode">Zip code*</label>
+				</div>
+				<span v-if="v$.address.zipCode.$error && submitted">
+					<span id="zipCode-error" v-for="(error, index) of v$.address.zipCode.$errors" :key="index">
+						<small class="p-error">{{ error.$message }}</small>
+					</span>
+				</span>
+			</div>
+		</div>
+		<Button type="submit" label="Register" class="mt-2" v-on:keyup.enter="handleSubmit" />
+	</form>
+</template>
+
+<script setup>
+import { reactive, ref } from "vue";
+import { email, required } from "@vuelidate/validators";
+import { useVuelidate } from "@vuelidate/core";
+import Button from "primevue/button";
+import InputText from "primevue/inputtext";
+import InputNumber from "primevue/inputnumber";
+import Password from "primevue/password";
+import Divider from "primevue/divider";
+import InputMask from "primevue/inputmask";
+import { register } from "@/services/api";
+
+const state = reactive({
+	firstname: "",
+	lastname: "",
+	pesel: null,
+	email: "",
+	password: "",
+	address: {
+		city: "",
+		zipCode: null,
+		street: "",
+		houseNumber: null,
+		localNumber: null,
+	},
+});
+
+const rules = {
+	firstname: { required },
+	lastname: { required },
+	pesel: { required },
+	email: { required, email },
+	password: { required },
+	address: {
+		city: { required },
+		zipCode: { required },
+		street: { required },
+		houseNumber: { required },
+		localNumber: {},
+	},
+};
+
+const submitted = ref(false);
+
+const v$ = useVuelidate(rules, state);
+
+const handleSubmit = (isFormValid) => {
+	submitted.value = true;
+	if (!isFormValid) {
+		return;
+	}
+	register(state.firstname, state.lastname, state.pesel, state.email, state.password, state.address)
+		.then(() => {
+			resetForm();
+		})
+		.catch((err) => {
+			console.error(err);
+		});
+};
+
```

Comment:
Myślę, że przydałoby nam się powiadomienie o prawidłowej rejestracji oraz powiadomienie o błędzie (podobnie jak mamy np. przy dodawaniu slotów). Dzięki temu użytkownik otrzymałby przede wszystkim informację zwrotną, czy wszystko poszło dobrze, a również informację co poszło nie tak po stronie backendu - gdzie przekazujemy komunikaty np. o istniejącym mailu, istniejącym numerze PESEL, czy nieprawidłowym numerze PESEL.

Reply:
Już dodane, zrobiłam na wzór powiadomienia z dodawania slotów.

Reply:
Super, dzięki! To zamykam 👍